src/test/rpc/AccountLinesRPC_test.cpp

```diff
@@ -524,11 +524,198 @@ class AccountLinesRPC_test : public beast::unit_test::suite
             RPC::make_error(rpcINVALID_PARAMS)[jss::error_message]);
     }
 
+    void
+    testAccountLinesWalkMarkers()
+    {
+        testcase("Marker can point to any appropriate ledger entry type");
+        using namespace test::jtx;
+        using namespace std::chrono_literals;
+        Env env(*this);
+
+        // The goal of this test is observe account_lines RPC calls return an
+        // error message when the SLE pointed to by the marker is not owned by
+        // the Account being traversed.
+        //
+        // To start, we'll create an environment with some trust lines, offers
+        // and a signers list.
+        Account const alice{"alice"};
+        Account const becky{"becky"};
+        Account const gw1{"gw1"};
+        env.fund(XRP(10000), alice, becky, gw1);
+        env.close();
+
+        // A couple of helper lambdas
+        auto escrow = [&env](
+                          Account const& account,
+                          Account const& to,
+                          STAmount const& amount) {
+            Json::Value jv;
+            jv[jss::TransactionType] = jss::EscrowCreate;
+            jv[jss::Flags] = tfUniversal;
+            jv[jss::Account] = account.human();
+            jv[jss::Destination] = to.human();
+            jv[jss::Amount] = amount.getJson(JsonOptions::none);
+            NetClock::time_point finish = env.now() + 1s;
+            jv[sfFinishAfter.jsonName] = finish.time_since_epoch().count();
+            return jv;
+        };
```

Comment:
I agree. I actually stole my code from Escrow_test.cpp, I think.

Reply:
https://github.com/XRPLF/rippled/pull/4396

Reply:
Hi @dangell7, thanks for the offer.  I can't give you a _precise_ definition of what I'm looking for, but here's a hand wave.

First, note that there are several different unit test files that locally define functions or lambdas which

1. Create escrows,
2. Finish escrows, or
3. Cancel escrows.

Here's where those functions are defined (locally) in Escrow_test.cpp: https://github.com/XRPLF/rippled/blob/develop/src/test/app/Escrow_test.cpp#L170-L190

Since this functions are only defined locally, they are not available to all of the test files that need to do these escrow operations.  So the functions have been copied, pasted, and mutated into different test files.  The `escrow` lambda, on line 548 here in this file, is one such example.

However, if we pulled those functions out of Escrow_test.cpp and put them somewhere they could be used by all the tests, then we can remove a lot of code duplication.

An example you could consider following is the src/test/jtx/check.h and src/test/jtx/impl/check.cpp files.  Those files create a `check` namespace and put all of the check-related functions in there.  So in tests the functions are invoked like `check::create(...)` or `check::cash(...)`.  That seems like a pretty good approach for less common facilities like checks and escrows.

But once you've made those functions in the `escrow` namespace the work has only started.  Now you need to go through all of the tests that use escrows and remove all of those local functions and lambdas.  Replace calls to those local functions will calls to the new functions you created in the `escrow` namespace.  This _may_ get tricky, because the escrow-related functions that have been copied around may have mutated.  So the interfaces of the copied functions may not be uniform across the code base.

Once you've...

1. Created the new globally accessible test functions,
2. Removed all of the local definitions,
3. Replaced calls to those local definitions with calls to the new global functions, _and_
4. The unit tests all pass.

Then you're done.

Does that make sense?  It may be more work than you want to take on in your pull request.  It may make more sense as a separate pull request.